src/shogun/lib/SGVector.cpp

```diff
@@ -310,12 +337,14 @@ void SGVector<T>::add(const SGSparseVector<T>& x)
 template<class T>
 void SGVector<T>::display_size() const
 {
+	on_gpu_check();
 	SG_SPRINT("SGVector '%p' of size: %d\n", vector, vlen)
 }
 
 template<class T>
 void SGVector<T>::copy_data(const SGReferencedData &orig)
 {
+	gpu_vector=std::shared_ptr<GPUMemoryBase<T>>(((SGVector*)(&orig))->gpu_vector);
```

Comment:
this crashes if I pass SGMatrix, no?
If so, this was a bug. Could need a fix there